src/main/java/com/microsoft/sqlserver/jdbc/ThreePartName.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class ThreePartName {
+    /*
+     * Three part names parsing For metdata calls we parse the procedure name into parts so we can use it in sp_sproc_columns sp_sproc_columns
+     * [[@procedure_name =] 'name'] [,[@procedure_owner =] 'owner'] [,[@procedure_qualifier =] 'qualifier']
+     *
+     */
+    private static final Pattern THREE_PART_NAME = Pattern.compile(JDBCSyntaxTranslator.getSQLIdentifierWithGroups());
+
+    private final String databasePart;
+    private final String ownerPart;
+    private final String procedurePart;
+
+    private ThreePartName(String databasePart, String ownerPart, String procedurePart) {
+        this.databasePart = databasePart;
+        this.ownerPart = ownerPart;
+        this.procedurePart = procedurePart;
+    }
+
+    public String getDatabasePart() {
```

Comment:
You could and it'd still compile (as it's in the same package as the usage). They're marked public as they're part of the public contract of that class.

If anything, the modifier for the ThreePartName class should be expanded from the default of package-private to public to allow usage from other packages within the driver source. There were not any such usages thus far, as the previous version was an inner class within SQLServerCallableStatement, I didn't mark it as such and figured that could be done when other code needs similar parsing.